cranelift/codegen/meta/src/gen_inst.rs

```diff
@@ -1104,20 +1107,42 @@ fn gen_isle(formats: &[&InstructionFormat], instructions: &AllInstructions, fmt:
     // Generate all the extern type declarations we need for various immediates.
     fmt.line(";;;; Extern type declarations for immediates ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;");
     fmt.empty_line();
-    let imm_tys: BTreeSet<_> = formats
+    let rust_name = |f: &FormatField| f.kind.rust_type.rsplit("::").next().unwrap();
+    let fields = |f: &FormatField| f.kind.fields.clone();
+    let imm_tys: BTreeMap<_, _> = formats
         .iter()
         .flat_map(|f| {
             f.imm_fields
                 .iter()
-                .map(|i| i.kind.rust_type.rsplit("::").next().unwrap())
+                .map(|i| (rust_name(i), fields(i)))
                 .collect::<Vec<_>>()
         })
         .collect();
-    for ty in imm_tys {
+    for ty in imm_tys.keys().filter(|&&k| k != "FloatCC") {
         fmtln!(fmt, "(type {} (primitive {}))", ty, ty);
     }
     fmt.empty_line();
 
+    // Generate `FloatCC` enumeration.
```

Comment:
Can you do this for all enum operands including `IntCC`?

Reply:
So `IntCC`, `AtomicRmwOp`, and `TrapCode` from [immediates.rs](https://github.com/bytecodealliance/wasmtime/blob/a7aa1723e0a00ab48a2cff304a8a8aa8b351c390/cranelift/codegen/meta/src/shared/immediates.rs#L160)?

Reply:
Yes